### PR TITLE
Add test cases for all axes types

### PR DESCRIPTION
**Pull request by stockiNail**
This PR is:

1. adding test cases for all axes types
2. adding `chartjs-adapter-luxon` and `luxon` dependencies (enabling tests on time axes)
3. adding `test-lint` mpn command in order to add the lint test invoking `npm test`

### CODE REVIEW

**Comment by stockiNail** [comment]
CI fails because "Set the SONAR_TOKEN env variable.". I don't see where you remove Windows test. Maybe I could change GHA ci file.

Furthermore I removed (temporarily) the test on time axes because CI failed (even if locally it works).

**Comment by kurkle** [comment]
Try to rebase to latest master, should help.

**Comment by kurkle** [comment]
hmm, no, its about pr from forked repo. I'll need to investigate a bit more.

**Comment by stockiNail** [comment]
> hmm, no, its about pr from forked repo. I'll need to investigate a bit more

yes because I didn't find anything new to merge.

**Comment by stockiNail** [comment]
> Furthermore I removed (temporarily) the test on time axes because CI failed (even if locally it works).

I think have found the reason... could be the "locale"

**Comment by kurkle** [comment]
This is the reason SONAR_TOKEN is not set:

> Note: With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository.

**Comment by stockiNail** [comment]
> Note: With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository.

Ok... that means the workflow will always fail, I guess.

File: test/fixtures/xCartesianTime.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+module.exports = {
+  config: {
+    type: 'line',
+    data: {
+      datasets: [{
+        data: [
+          {x: 1648249200000, y: 24},
```
Comment (kurkle) [inline test/fixtures/xCartesianTime.js:7]: You should use strings here without timezone, so it will be parsed and displayed at the same timezone

**Comment by kurkle** [comment]
> > Note: With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository.
> 
> Ok... that means the workflow will always fail, I guess.

I updated the CI again to only run sonar if the token exists

**Comment by stockiNail** [comment]
> I updated the CI again to only run sonar if the token exists

Shall I rebase, I guess?

File: test/fixtures/xCartesianTime.js (hunk 1 above)
Comment (stockiNail) [reply test/fixtures/xCartesianTime.js]: I'm testing

File: test/fixtures/xCartesianTime.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+module.exports = {
+  config: {
+    type: 'line',
+    data: {
+      datasets: [{
+        data: [
+          {x: '26/03/2022', y: 24},
+          {x: '27/03/2022', y: 52},
+          {x: '28/03/2022', y: 79},
```
Comment (stockiNail) [inline test/fixtures/xCartesianTime.js:9]: Added as strings but it seems it's not working...

**Comment by kurkle** [comment]
Don't try to fix the duplication in tests :)

**Comment by stockiNail** [comment]
> Don't try to fix the duplication in tests :)

Already done... changed the exclusion in the `sonar.properties` (the paths must be comma separated but without blanks)

**Comment by kurkle** [comment]
The exclusions only apply when analysis is run from CI. This sonarcloud bot analysis only uses the settings I provided from sonarcloud UI (and it was missing the exclusions).
I'm not sure if the spaces are needed for the CI.

**Comment by stockiNail** [comment]
> This sonarcloud bot analysis only uses the settings I provided from sonarcloud UI (and it was missing the exclusions).
> I'm not sure if the spaces are needed for the CI.

Anyway ok for me! Tomorrow (if I can, or at least on Monday) I will have a look to Sonar coverage because it's not clear to me what I have seen.

**Comment by stockiNail** [comment]
And I need to understand why the tests on time don't work, even if I used the dates in the data as strings. But I can submit another PR for those tests

**Comment by kurkle** [comment]
can you rebase once more? I'd like to see if I can get the coverage reported here :)

**Comment by stockiNail** [comment]
> can you rebase once more? I'd like to see if I can get the coverage reported here :)

okI!!

**Comment by stockiNail** [comment]
> can you rebase once more? I'd like to see if I can get the coverage reported here :)

DONE!

**Comment by kurkle** [comment]
ok, one more? :)

**Comment by kurkle** [comment]
Ok, fine. I'll explore more options later.

**Comment by stockiNail** [comment]
> Ok, fine. I'll explore more options later.

OK! apologize if I could fix it :(
I didn't want to add work to you.

**Comment by kurkle** [comment]
> > Ok, fine. I'll explore more options later.
> 
> OK! apologize if I could fix it :( I didn't want to add work to you.

No worries, I need to learn anyway how to best use sonarcloud.

**Pull request merged**